### PR TITLE
applied optionT if it is possible

### PR DESCRIPTION
**Pull request by danieljimenezr**

### CODE REVIEW

File: src/test/scala/com/lambtors/poker_api/module/poker/behaviour/PokerBehaviourSpec.scala [hunk 1]
```diff
@@ -21,10 +24,10 @@ trait PokerBehaviourSpec
   protected val playerRepository: PlayerRepository[Future]       = mock[PlayerRepository[Future]]
 
   def shouldFindPokerGame(gameId: GameId, pokerGame: PokerGame): Unit =
-    (pokerGameRepository.search _).expects(gameId).once().returning(Future.successful(Some(pokerGame)))
+    (pokerGameRepository.search _).expects(gameId).once().returning(OptionT.fromOption[Future](Some(pokerGame)))
```
Comment (GMadorell) [inline src/test/scala/com/lambtors/poker_api/module/poker/behaviour/PokerBehaviourSpec.scala]: maybe there is some method like optiont.none but for a some?

File: src/main/scala/com/lambtors/poker_api/module/poker/application/flop/FlopCardsAdder.scala [hunk 2]
```diff
@@ -13,22 +13,20 @@ final class FlopCardsAdder[P[_]: MonadErrorThrowable](
...
-            playerRepository
-              .search(gameId)
-              .flatMap(
-                players =>
-                  repository.update(
-                    game.copy(tableCards = deckProvider.shuffleGivenDeck(availableCards(players)).take(3))
-                )
+  def add(gameId: GameId): P[Unit] = {
+    repository.search(gameId).fold[P[Unit]](MonadErrorThrowable[P].raiseError(PokerGameNotFound(gameId)))(game =>
+      thereAreCardAtTheTable(game.tableCards).ifM(
```
Comment (GMadorell) [inline src/main/scala/com/lambtors/poker_api/module/poker/application/flop/FlopCardsAdder.scala]: thereAreCardAtTheTable => existsCardAtTable

File: src/main/scala/com/lambtors/poker_api/module/poker/application/player_cards/find/PlayerCardsFinder.scala [hunk 3]
```diff
@@ -10,6 +10,7 @@ final class PlayerCardsFinder[P[_]: MonadErrorThrowable](playerRepository: Playe
   def find(playerId: PlayerId): P[(Card, Card)] =
     playerRepository
       .search(playerId)
-      .flatMap(_.fold[P[(Card, Card)]](MonadErrorThrowable[P].raiseError(PlayerNotFound(playerId)))(player =>
-        (player.firstCard, player.secondCard).pure[P]))
+      .fold[P[(Card, Card)]](MonadErrorThrowable[P].raiseError(PlayerNotFound(playerId)))(player =>
+        (player.firstCard, player.secondCard).pure[P])
```
Comment (GMadorell) [inline src/main/scala/com/lambtors/poker_api/module/poker/application/player_cards/find/PlayerCardsFinder.scala]: (player.firstCard, player.secondCard) => player.cards (method that returns a tuple inside the case class)?

File: src/main/scala/com/lambtors/poker_api/module/poker/application/river/RiverCardAdder.scala [hunk 4]
```diff
@@ -19,26 +19,26 @@ class RiverCardAdder[P[_]: MonadErrorThrowable](
...
-                      game.copy(
-                        tableCards = game.tableCards ++ deckProvider
-                          .shuffleGivenDeck(availableCards(playersCards(players) ++ game.tableCards))
-                          .take(1)
-                      )
-                  )
+      .fold[P[Unit]](MonadErrorThrowable[P].raiseError(PokerGameNotFound(gameId)))(game =>
+        cardsAtTableNumberGreaterThanFour(game.tableCards).ifM(
+          MonadErrorThrowable[P].raiseError(RiverNotPossibleWhenItIsAlreadyGiven(gameId)),
+          cardsAtTableNumberLowerThanFour(game.tableCards).ifM(
```
Comment (GMadorell) [inline src/main/scala/com/lambtors/poker_api/module/poker/application/river/RiverCardAdder.scala]: existsLessThanFourCardsAtTable

File: src/main/scala/com/lambtors/poker_api/module/poker/application/river/RiverCardAdder.scala [hunk 5]
```diff
@@ -19,26 +19,26 @@ class RiverCardAdder[P[_]: MonadErrorThrowable](
...
-                  players =>
-                    repository.update(
-                      game.copy(
-                        tableCards = game.tableCards ++ deckProvider
-                          .shuffleGivenDeck(availableCards(playersCards(players) ++ game.tableCards))
-                          .take(1)
-                      )
-                  )
+      .fold[P[Unit]](MonadErrorThrowable[P].raiseError(PokerGameNotFound(gameId)))(game =>
+        cardsAtTableNumberGreaterThanFour(game.tableCards).ifM(
```
Comment (GMadorell) [inline src/main/scala/com/lambtors/poker_api/module/poker/application/river/RiverCardAdder.scala]: using simple ifs we have less nested code than with ifM, does using ifM still make sense?

**Review by GMadorell — APPROVED** [review]

File: src/main/scala/com/lambtors/poker_api/module/poker/application/river/RiverCardAdder.scala (hunk 5 above)
Comment (danieljimenezr) [reply src/main/scala/com/lambtors/poker_api/module/poker/application/river/RiverCardAdder.scala]: use ifM helps us to have more declarative code, read it!

**Pull request merged**